src/syntax_highlight.rs

```diff
@@ -34,7 +36,8 @@ unsafe impl Sync for Setup {}
 lazy_static!{
     static ref SETUP: Setup = Setup {
         syntax_set: SyntaxSet::load_defaults_newlines(),
-        theme_set: ThemeSet::load_defaults()
+        theme_set: ThemeSet::load_defaults(),
+        theme_name: RefCell::new(DEFAULT_THEME_NAME.to_owned()),
```

Comment:
I found a way with closures.  You can [view the alternate patch set](https://github.com/boxofrox/cobalt.rs/commits/config-theme-closure-example) or [compare against this PR](https://github.com/boxofrox/cobalt.rs/compare/config-theme...boxofrox:config-theme-closure-example).  With closures I was able to drop the reversion path and `RefCell`.

Let me know if the alternate is preferred and I'll swap it into this PR.